src/frontend/screens/MapScreen/TrackBottomSheet/index.tsx

```diff
@@ -19,18 +19,44 @@ export const TrackBottomSheet = React.memo(() => {
   const [backgroundPermission, setBackgroundPermission] =
     React.useState<Location.LocationPermissionResponse | null>(null);
 
+  const checkPermissions = React.useCallback(async () => {
+    const [foreground, background] = await Promise.all([
+      Location.getForegroundPermissionsAsync(),
+      Location.getBackgroundPermissionsAsync(),
+    ]);
+    setForegroundPermission(foreground);
+    setBackgroundPermission(background);
+  }, []);
+
+  // Re-check permissions on screen focus
+  // Handles re-checking when navigating back from in-app
   useFocusEffect(
     React.useCallback(() => {
-      Location.getForegroundPermissionsAsync().then(permission =>
-        setForegroundPermission(permission),
-      );
-
-      Location.getBackgroundPermissionsAsync().then(permission =>
-        setBackgroundPermission(permission),
-      );
-    }, []),
+      checkPermissions();
```

Comment:
I don't think we need this useCallback anymore as checkPermissions is already memoized